### PR TITLE
Add Inital Novu Architecture tab

### PR DESCRIPTION
**Pull request by Cliftonz**
Adds the initial architecture overview for Novu

### CODE REVIEW

File: architecture/diagrams.mdx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+title: "Novu Architecture Diagram"
+---
+# Novu Architecture Diagram powered by [terrastruct](https://terrastruct.com) and D2
+
+If the diagram below is not rendering, please [click here](https://app.terrastruct.com/diagrams/1895866270) to view it on terrastruct.com.
```
Comment (unicodeveloper) [inline architecture/diagrams.mdx:6]: So does this diagram capture version 0.21. or 0.22? @Cliftonz

File: architecture/introduction.mdx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+Our systems are hosted on Amazon Web Services (AWS), ensuring a reliable and secure environment for our infrastructure.
+We adhere to several industry-standard compliance certifications including SOC II, PCI, and GDPR which underline our commitment to security and data privacy.
+
+## Framework Leveraged
+
+- Nestjs: A progressive Node.js framework that is used for building efficient, reliable, and scalable server-side applications. Nestjs provides a high-level abstraction with a lot of out-of-the-box features, making it a perfect choice for building modular and maintainable applications.
+- Mongoose: An elegant Object Document Mapper (ODM) for MongoDB, which simplifies the interaction with the database by providing a straightforward modeling of the application data. Mongoose’s features like schema validation, query building, and middleware support immensely help in building robust applications.
+- BullMQ: A robust, fast, and reliable queue library for handling distributed jobs and messages in Node.js. It is instrumental in managing the processing of notification jobs ensuring they are handled in an efficient and reliable manner.
```
Comment (unicodeveloper) [inline architecture/introduction.mdx:15]: Please make `Nestjs`, `Mongoose`, `BullMQ` bold so it's differentiated from the detailed explanation of each

File: architecture/introduction.mdx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+## Databases
+
+- MongoDB: A NoSQL database that provides high performance, high availability, and easy scalability. It works on a document-oriented data model, which is extremely flexible and allows for varied, complex data structures to be stored and manipulated with ease.
+- Redis: An in-memory data structure store, used as a database, cache, and message broker. It's exceptionally fast and provides support for various data structures like strings, hashes, lists, sets, etc. It's utilized in Novu for caching and as a reliable message broker for managing queues and real-time data processing.
```
Comment (unicodeveloper) [inline architecture/introduction.mdx:20]: Please make `MongoDB`, and `Redis` bold so it's differentiated from the detailed explanation of each

File: architecture/diagrams.mdx (hunk 1 above)
Comment (Cliftonz) [reply architecture/diagrams.mdx:6]: This document is a rolling document at the moment, currently it shows 0.21 but will be updated soon for 0.22

Comment (unicodeveloper) [reply architecture/diagrams.mdx:6]: Okay, so you will need to version this page. you can rename it to `diagrams-021.mdx`. 

and then add version like this:

```
---
title: "Novu Architecture Diagram"
version: 'v0.21.0'
---
```

@Cliftonz

**Comment by unicodeveloper** [comment]
Looking forward to the versioning @Cliftonz

**Comment by Cliftonz** [comment]
@unicodeveloper  pushed

**Review by unicodeveloper — APPROVED** [review]

**Pull request merged**